pkg/cas/resolver/resolver.go

```diff
@@ -69,13 +89,63 @@ func (h *Resolver) Resolve(webCASURL *url.URL, cid string, data []byte) ([]byte,
 			}
 		}
 
-		return nil, fmt.Errorf("failed to get data stored at %s from the local CAS: %w", cid, err)
+		if len(cidParts) > 1 {
+			return h.resolveCIDWithHint(cidParts)
+		}
+
+		if h.ipfsReader != nil {
+			// last resort - try ipfs if reader configured
+			return h.getAndStoreDataFromIPFS(localCID)
+		}
+
+		return nil, fmt.Errorf("failed to get data stored at %s from the local CAS: %w", localCID, err)
 	}
 
 	return dataFromLocal, nil
 }
 
-func (h *Resolver) getAndDataFromRemote(webCASEndpoint *url.URL, cid string) ([]byte, error) {
+func (h *Resolver) resolveCIDWithHint(cidParts []string) ([]byte, error) {
+	var dataFromRemote []byte
+
+	switch cidParts[0] {
+	case "webcas":
+		cid := cidParts[2]
+		domain := cidParts[1]
+
+		// TODO: fix hard-coded URL when webfinger is available
```

Comment:
Can you add (#364) to this TODO to help me remember to change this?

Reply:
Done